### DISCUSSION TITLE
Confused about Graph Connector and PnP Search Result Column Mappings

### DISCUSSION THREAD

**Discussion by renewtx** [microsoft-search/pnp-modern-search#2964] [Q&A] [answered]
Hi,
I have created a Data Source in M365  that uses the Graph File Share connector.  It is successfully connecting  and indexing content, and content based on that connector is surfaced in Microsoft Search ... 
![image](https://github.com/microsoft-search/pnp-modern-search/assets/58831734/5deff2b7-1224-46ce-aa6b-314001f20ebb)

When I look at the connector/data source configuration, it shows the following (out-of-the-box, default) label/column mappings:
![image](https://github.com/microsoft-search/pnp-modern-search/assets/58831734/ad2a8277-d2b2-4976-baca-3252a2e03d11)

I assume that these are the properties/columns that the connector supports out-of-the-box when indexing file share content.

Now I want to set up a PnP Search page that queries against this content and so I set up a search result web part that uses the Microsoft Search data source, uses External Items as the Entity Types to search,  and the File Share Connector as the Content Source.
![image](https://github.com/microsoft-search/pnp-modern-search/assets/58831734/e829539d-d13f-4a57-81d1-559104c3fee0)

I was able to map the Name, Created, and Source columns and i get results as expected.
![image](https://github.com/microsoft-search/pnp-modern-search/assets/58831734/bbe3c5f1-c55b-4e15-b497-da10da24abd8)

What I would like to include in my result grid, however, is the URL to the files and I cant seem to access that.  When I click on the Selected Fields control I see this list
![image](https://github.com/microsoft-search/pnp-modern-search/assets/58831734/bf2d8cc3-5e31-4c3e-b927-d18e4e598932)

and only some of these match what the connector shows above ... how does PnP Search determine the properties to show on the Selected Fields list?  WHere are those properties coming from?  For example, how do I get the URL to the item (which is obviously in the index since the default MS Search can display it)?

Second, where does the list of Slot Fields come from?  When I click the Slot Field dropdown, I see this
![image](https://github.com/microsoft-search/pnp-modern-search/assets/58831734/dbb4911b-772a-4f06-a9a9-0d9a318c87b0)

which again does not seem to match what is available in the Selected Fields list ... 

Can anyone shed some clarifications on this?  Using PnP Search v4.8.

Thanks

**Comment by kasperbolarsen** [comment]
We recommend using either SP Editor (Chrome Extension) or the SharePoint Search Query tool for verifying the data we are working with,  when SharePoint is the data source. 
When the data source is MS Search we recommend using the Graph Explorer (the search segment) to verify the query and the properties that is returned.

**Comment by renewtx** [comment]
Not sure what you mean by "verify the query and the properties that is returned." In my case the searches are working fine ... what I dont understand is where does PnP Search get the list of properties that appear in the Selected Fields control, and where does the Slot Fields list get its values?  This is is pure PnP Search functionality and has nothing to do with me running a query, correct?  

I apologize if I am not seeing something, but as a non-developer I am simply trying to use PnP Search to build a search result and add proper columns to my result grid, but its difficult to do this without knowing where the above controls are getting their values and esp since none seem to match ... does that make sense?

**Reply by renewtx** [reply, marked as answer]
I think I just answered my own question (at least partially) ... I was not aware that I could manually add values to the Selected Fields control ...  once I manually entered the value of 'url' into the text field and saved it, that value was now available on the Slot Field dropdown for me to select and i could then add it to the grid via the Manage Columns screen, and now I am able to display the URL to the files ... that just wasn't 'intuitively obvious' at first ... 
![image](https://github.com/microsoft-search/pnp-modern-search/assets/58831734/781fdadc-c2da-4d4d-86c7-a5f923b54ed8)

I am still curious, however, where/how does PnP Search determine the initial set of values to show on the Selected Fields list?  What is that initial list based on?
